File: Server/src/main/content/region/misthalin/quest/member/gertrudescat/npc/FluffNPC.java
package content.region.misthalin.quest.member.gertrudescat.npc;

import core.game.node.entity.npc.AbstractNPC;
import core.game.node.entity.player.Player;
import core.game.world.map.Location;
import core.plugin.Initializable;

/**
 * The Fluff npc.
 */
@Initializable
public final class FluffNPC extends AbstractNPC {
    private static final int[] ID = {2997};

    /**
     * Instantiates a new Fluff npc.
     */
    public FluffNPC() {
        super(0, null);
    }

    private FluffNPC(int id, Location location) {
        super(id, location);
    }

    @Override
    public AbstractNPC construct(int id, Location location, Object... objects) {
        return new FluffNPC(id, location);
    }

    @Override
    public boolean isHidden(final Player player) {
        if (player.getQuestRepository().getQuest("Gertrude's Cat").getStage(player) < 20) {
            return true;
        }
        return player.getAttribute("hidefluff", 0L) > System.currentTimeMillis();
    }

    @Override
    public int[] getIds() {
        return ID;
    }

}

File: Server/src/main/content/region/misthalin/quest/member/gertrudescat/npc/FluffNPC.kt
package content.region.misthalin.quest.member.gertrudescat.npc

import core.api.consts.NPCs
import core.game.node.entity.npc.AbstractNPC
import core.game.node.entity.player.Player
import core.game.world.map.Location
import core.plugin.Initializable

/**
 * Fluff NPC.
 */
@Initializable
class FluffNPC(id: Int = 0, location: Location? = null) : AbstractNPC(id, location) {

    // Constructor for FluffNPC class.
    override fun construct(id: Int, location: Location, vararg objects: Any): AbstractNPC {
        return FluffNPC(id, location)
    }

    /**
     * Checks if fluff is hidden for the player.
     *
     * @param player the player to check.
     * @return true if fluff is hidden, false otherwise.
     */
    override fun isHidden(player: Player): Boolean {
        if (player.getQuestRepository().getQuest("Gertrude's Cat").getStage(player) < 20) {
            return true // Hide fluff if player's quest stage is less than 20.
        }
        return player.getAttribute("hidefluff", 0L) > System.currentTimeMillis() // Check if fluff should be hidden based on time.
    }

    // Get the NPC IDs.
    override fun getIds(): IntArray {
        return ID
    }

    companion object {
        private val ID = intArrayOf(NPCs.GERTRUDES_CAT_2997) // Array of NPC IDs.
    }
}
